pretyping/indrec.ml

```diff
@@ -601,15 +606,17 @@ let mis_make_indrec env sigma ?(force_mutual=false) listdepkind mib u =
 
 let build_case_analysis_scheme env sigma pity dep kind =
   let specif = lookup_mind_specif env (fst pity) in
-  if dep && not (Inductiveops.has_dependent_elim specif) then
-    raise (RecursionSchemeError (env, NotAllowedDependentAnalysis (false, fst pity)));
   mis_make_case_com dep env sigma pity specif kind
 
 let is_in_prop mip =
   match inductive_sort_family mip with
   | InProp -> true
   | _ -> false
 
+let default_case_analysis_dependence env ind =
+  let _, mip as specif = lookup_mind_specif env ind in
+  not (is_in_prop mip || not (Inductiveops.has_dependent_elim specif))
+
 let build_case_analysis_scheme_default env sigma pity kind =
   let _, mip as specif = lookup_mind_specif env (fst pity) in
   let dep = not (is_in_prop mip || not (Inductiveops.has_dependent_elim specif)) in
```

Comment:
Shouldn't this be shared `with default_case_analysis_dependence`?

Reply:
I don't care if we fetch the specif twice or if we make `default_case_analysis_dependence` take a specif argument, I just feel the boolean computation should have a single source of truth by being shared, because it looks like a kinda adhoc boolean.

Reply:
Done, if you insist.